// components/Snippet/Snippet.js
import PropTypes            from 'prop-types'
import React                from 'react'

import { Segment }          from 'semantic-ui-react'

import Editor               from '../Editor'
import Flex                 from '../Flex'

class Snippet extends React.Component {
  static propTypes = {
    description: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    path: PropTypes.string.isRequired,
    theme: PropTypes.object.isRequired,
  }

  getCode() {
    const { path } = this.props
    return require(`!raw-loader!../../examples/${ path }`)
  }

  render() {
    const { description, name, path, theme } = this.props

    return (
      <Segment style={{ width: 600 }}>
        <Flex nowrap style={{ border: `1 solid #ffff` }}>
          <div style={{ width: `50%`, padding: 10 }}>
            <Flex center middle style={{ height: 200 }}>
              { require(`../../examples/${ path }`).default({ theme }) }
            </Flex>
          </div>

          <div style={{ width: `50%`, padding: 10 }}>
            <div><strong>{ name }</strong></div>
            <div>{ description }</div>
          </div>
        </Flex>
        <Flex>
          <Editor
            id={ name }
            value={ this.getCode() }
          />
        </Flex>
      </Segment>
    )
  }
}

export default Snippet
